### DISCUSSION TITLE
Vaccinations - best practice?

### DISCUSSION THREAD

**Discussion by Pletoo** [sheltermanager/asm3#1518] [Q&A / Help]
Typically the Distemper vaccine is given to all intakes within their 1st week and kittens can get their 1st anywhere between 6-10 weeks of age. I have created species and baby templates to add that initial shot to the vaccine calendar so that we can use the mobile UI to quickly mark that vaccine as given when the time comes. (We administer this shot ourselves instead of taking them to the vet.)

However if the cat is pregnant, nursing, sick, or underweight that shot can be delayed as much as 4-6 weeks. There are times when that is 50+ vaccines that are alerting as due today that can't be given and the medical calendar and vaccine book get very messy. It also means that I am scrolling through quite a list on my phone and it can be hard to pick out the ones that are actually able to get their shot. 

Kittens have to get a booster 3-4 weeks after the 1st and I make note of that using the comments when I schedule the booster so I can sort the comment column and find them all in the vaccine book.  I know I can go in and bulk change the due dates for any that are not ready yet, I don't use the vaccination book or templates at all for Rabies, I can add and give them through a CSV import along with everything else from vet day, so that cleaned up the book and alerts somewhat...

...but I was wondering if there were a better way that I have overlooked that doesn't involve using both the laptop and phone at the same time on vaccine day (or switching between interfaces?)

I know it isn't currently possible, but I think it might be handy if we could mark a vaccine as postponed and have it show up in the vaccine book as a different color and not be counted in the same alert as the ones that are actually due (kitten boosters and healthy intakes.) Then perhaps a filter and an option to collapse the postponed ones on the mobile UI.

**Comment by bobintetley** [comment]
This seems overly complicated. Could you not just move the required date on the vaccination record when it is postponed? The vaccine book should make it easy to edit the due ones quickly.

**Reply by Pletoo** [reply]
Yes. I can just change the required date. Right now that could be as many as 75 dates changed - and next week I will have to change most of them again. 
Right now in addition to adult cats, we have:

- 38 kittens that are anywhere from 5-12 weeks that have not had a Distemper shot yet. 
- 42 that are 13-20 weeks old, some have not had their 1st shot but others have and need their booster within 3-4 weeks or have to start over. (sometimes that 2nd shot is given when altered, sometimes it is not.)
- 4 nursing mothers that can't get shots until kittens are weaned (but that could be anytime)

We have rules and limitations such as:

- first Distemper _can_ be given at 6 weeks
- first Distemper **must** be given at least a week before going to the vet
- first vet visit is usually for spay/neuter, kitten has to be at least 3 months, 3 pounds
- we can give Distemper shots at the center, but have a limited weekly supply, and the date is based on when those that administer the shot are available.
- We usually only have one vet day a week, with a limited number of patient slots to divide between kittens and any new adult intakes, tests, exams....

I can know in advance which ones we _plan_ on taking each week, but invariably we have:

- a sick or injured cat that takes priority
- a kitten that was scheduled but is sick or underweight and another is substituted.
- sometimes a fosterer brings in the wrong kitten

In addition - Although our paper records are in excellent order - the same can't be said for our SM vaccine book - because my co-worker spent 4 years working on her own and not realizing that vaccines needed 2 dates to be counted as given not just required. I have spent weeks cleaning that up, and old habits being hard to break, I still have had to go in and compare records to make sure that wasn't still happening. I tried to use the templates but those needed to rely on entry type being correct, another hard habit to break after 4 years of defaulting to stray. Also some shots were getting duplicated between manual entry and the template. 

Fortunately, after months of shared access, it was finally (two days ago) agreed that I should be the only one putting in vaccine information in to SM through the desktop interface. 

So yes. I agree. This all sounds overly complicated, but I assume you were talking from a coding standpoint.

Perhaps my idea would be difficult, but maybe there are other ways to achieve at least some of these? (the first is not as relevant now that I am _hopefully_ the only one maintaining the records, but I can't be the only one who could use a little help tracking such a flexible thing like this vaccine.)

- A way to be reasonably certain I was the one who added that vaccine or had verified it (pretty sure my co-worker would not have used a postpone feature when she wasn't even aware the vaccine book existed, she always did vaccines through individual records.)
- A reminder that a shot is due without having a red alert, that is more flexible than the current system.
- **The big one:** A way to use the mobile interface to record shots given without scrolling through 100 potential names, my thought was that a "postponed" shot would be in a collapsed menu, still accessible if there was a last minute change but otherwise out of the way.

**Comment by bobintetley** [comment]
You can tell who added or last changed a vaccination - click on it in the list to open the dialog, and there's an icon at the bottom left that looks like two little heads. Click this button to see who added and last changed that vacc.

It's also possible to use the vaccination book to bulk change the required date on all the selected vaccinations in one go.

**Reply by Pletoo** [reply]
So to answer my original question...is there a better way?

Basically, for me, I've decided that it is easier to skip using the vaccine book or templates for new vaccines. 
Instead, I am going to use a spreadsheet and a CSV import when 1st Distemper shots are given.
Then I can use the book strictly for scheduled boosters, since they are not as flexible.
I already do this for Rabies. I had just wanted to be able to use the mobile UI for recording1st Distemper.
That just means one less permission and a little more paperwork that I need to give my Fosterers.

Maybe once we are out from under kitten season and our numbers are more reasonable, I can look at it again.

**Reply by Pletoo** [reply]
Would it be possible to be able to also select animals from the vaccine book for the bulk vaccine?
I was hoping to just select from the ones given in the last week to schedule the booster rather than picking them from the full list on the add screen.
![image](https://github.com/user-attachments/assets/cc0968c1-ada8-4860-b6f8-3fd5c4e85fb1)
![image](https://github.com/user-attachments/assets/38fefea7-e32a-4efb-93c0-5243fd542eda)
